src/System.Net.Http/src/Internal/Mail/MailBnfHelper.cs

```diff
@@ -330,9 +333,21 @@ internal static string GetDateTimeString(DateTime value, StringBuilder builder)
                 localBuilder.Append(' ');
             }
 
-            string[] offsetFields = offset.Split(':');
-            localBuilder.Append(offsetFields[0]);
-            localBuilder.Append(offsetFields[1]);
+            int colonIndex = offset.IndexOf(':');
+            if (colonIndex == -1)
+            {
+                throw new InvalidOperationException("TimeSpan.ToString() should contain at least one colon.");
+            }
+
+            int nextColonIndex = offset.IndexOf(':', colonIndex + 1);
+            if (nextColonIndex == -1)
+            {
+                // Set it past the end of the string
+                nextColonIndex = offset.Length;
```

Comment:
That said, I don't know that it actually matters, as we're parsing the output from something in the framework, and these exceptions shouldn't happen.  Maybe asserts would be sufficient.  @davidsh, what do you want to do here?